index.js

```diff
@@ -1,5 +1,78 @@
 'use strict';
 
+var currentEventPortals = {};
+var MAX_EVENTS = 7;
+var X_CENTER = -10;
+var Y_CENTER = 11;
+var RADIUS = 4;
+
+function pollEvents() {
+    var xhr = new XMLHttpRequest();
+
+    xhr.open('GET', "https://account.altvr.com/api/home/recommended");
+    xhr.withCredentials = true;
+    xhr.onreadystatechange = function(e) {
+        if (this.readyState === XMLHttpRequest.DONE) {
+            var response = JSON.parse(this.response);
+            var sids = response.items.map(function(el) { return el.item.space.space_sid; });
+            var currentSids = Object.keys(currentEventPortals);
+
+            for (var sid of currentSids) {
+                if (sids.includes(sid)) continue;
+
+                var app = document.getElementById("app");
+                app.removeChild(currentEventPortals[sid]);
+                delete currentEventPortals[sid];
+            }
+
+            for (var i = 0; i < MAX_EVENTS; i++) {
+                if (!sids[i]) break;
+
+                var result = calculatePortalPositionAndRotation(i);
+                var position = result[0];
+                var rotation = result[1];
+
+                var existingPortal = currentEventPortals[sids[i]];
+                if (existingPortal) {
+                    existingPortal.setAttribute("position", position);
+                    existingPortal.setAttribute("rotation", rotation);
+                    continue;
+                }
+
+                var newPortal = document.createElement("a-entity");
+                newPortal.setAttribute("n-portal", "targetSpace: " + sids[i]);
+                newPortal.setAttribute("position", position);
+                newPortal.setAttribute("rotation", rotation);
+
+                var eventTitle = document.createElement("a-entity");
+                eventTitle.setAttribute("n-text", "text: " + response.items[i].item.name + "; fontSize: 2; width: 2;");
+                eventTitle.setAttribute("position", "0 0.3 0");
+                newPortal.appendChild(eventTitle);
+
+                document.getElementById("app").appendChild(newPortal);
+            }
+        }
+    };
+    xhr.send();
+
+    window.setTimeout(pollEvents, 5 * 60 * 1000);
+}
+
+// Center of circle is located at (-10, 0.5, 11), and the radius of it is 4 units.
+// Event portals are laid out from "right" to "left".
+function calculatePortalPositionAndRotation(i) {
+    var theta = i * 30;
+    var angle = (theta + 90) * (Math.PI / 180); // Adding 90 degrees here because we want to start laying out portals on the +y axis.
+    var xPos = RADIUS * Math.cos(angle) + X_CENTER;
+    var yPos = Y_CENTER - RADIUS * Math.sin(angle);
+
+    var position = xPos + " 0.5 " + yPos;
+    var rotation = "0 " + theta + " 0";
+    return [position, rotation];
+}
+
+window.onload = pollEvents;
```

Comment:
Reason not to use DOMContentLoaded or register the event rather than assigning to onload? Seems like this could potentially cause issues if something else is trying to bind to onload